confgenerator/confgenerator_test.go

```diff
@@ -85,9 +86,23 @@ func testJoin(goos string, elem ...string) string {
 	return strings.Join(elem, separator)
 }
 
+func overrideFilepathJoin(t *testing.T) {
+	if filepath.Join("a", "b", "c") != filepathJoin(runtime.GOOS, "a", "b", "c") {
+		t.Fatalf("default filepathJoin behaves unexpectedly")
+	}
+	filepathJoin = testJoin
```

Comment:
You're still updating a global. The test framework is free to schedule all of the goroutines you spawn within these tests first, and then execute all of them at once. This may be ok for the tests spawned in this function, because they all expect to run with the above implementation, but suppose another test is added in the future in a function that forgets to assign this. Not only would its behavior be platform-dependent, but it would also be randomly platform dependent (e.g., if this test function ran in the middle of executing the goroutines from the other function). Why not just use the above implementation in the actual config generator?

Reply:
My motivation was to make use of the real `filepath.Join` "in the wild", since it's more robust than `strings.Join(elem, separator)`. 

In any case, that was my rationale, but this is your project. I'll happily make the change as you've suggested.

Reply:
No worries. There's an argument to be made either way.

I switched it back and performed the override in `init`. 

One very minor repercussion: In order to pass the code coverage requirement, I need to test the default `filepathJoin` (or test something entirely unrelated). Since `init` overrides the default before there's a chance to test the code, I named it and called it directly in a test. It almost feels silly, but I tried to make the test at least somewhat useful in that it _may_ detect an unintended change to the default.